### ISSUE TITLE
JUPYTERHUB_API_URL doesn't respect hub_connect_ip or hub_connect_port

### ISSUE THREAD

**Issue by johnsca** [jupyterhub/kubespawner#211]
The `JUPYTERHUB_API_URL` env var passed to the spawned container is [set by the base `Spawner` class](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/spawner.py#L611) using `self.hub.api_url` which doesn't respect the `hub_connect_ip` or `hub_connect_port` settings.  It also can't be overridden with the `environment` setting because it's set after the `environment` setting is handled.

It seems this can be worked around using the `--hub-api-url` option, but it would be good to correct the `JUPYTERHUB_API_URL` environment variable as well.

**Comment by minrk** [comment]
hub_connect_ip and hub_connect_port should not be set on KubeSpawner, instead set on `JupyterHub` itself. KubeSpawner shouldn't have this duplicate config anymore. I think as it is right now, you may have to set *both* to get the right behavior with KubeSpawner, until it deprecates its `hub_connect_` config.

**Comment by minrk** [comment]
I just looked through the code, and I believe it should do the right thing *as long as you don't set* `c.KubeSpawner.hub_connect_ip`. If, instead, you set `c.JupyterHub.hub_connect_ip`, everything will be correct.

**Comment by minrk** [comment]
@elgalu I don't think so. The fix is to set the `c.JupyterHub.connect_ip|url` config directly, which should be source of this value everywhere. It doesn't need to be set in multiple places.
